sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -206,15 +207,35 @@ def _infer_event_timestamp_from_dataframe(entity_df: pandas.DataFrame) -> str:
 
 
 class BigQueryRetrievalJob(RetrievalJob):
-    def __init__(self, query, client):
+    def __init__(self, query, client, config):
         self.query = query
         self.client = client
+        self.config = config
 
     def to_df(self):
         # TODO: Ideally only start this job when the user runs "get_historical_features", not when they run to_df()
         df = self.client.query(self.query).to_dataframe(create_bqstorage_client=True)
         return df
 
+    def to_bigquery(self, dry_run=False) -> str:
+        path = f"{self.client.project}.{self.config.offline_store.dataset}.tmp-{str(uuid.uuid4())}"
+        job_config = bigquery.QueryJobConfig(destination=path, dry_run=dry_run)
+        bq_job = self.client.query(self.query, job_config=job_config)
+
+        if dry_run is True:
+            print("This query will process {} bytes.".format(bq_job.total_bytes_processed))
+            return
+
+        while True:
+            query_job = self.client.get_job(bq_job.job_id)
+            if query_job.state in ["PENDING", "RUNNING"]:
+                print(f"The job is still '{bq_job.state}'. Will wait for 30 seconds")
+                time.sleep(30)
```

Comment:
Is there anything preventing us from using the existing QueryJob functionality to wait for the job (like `result()` and `done()`)?

Reply:
see above:)